app/src/main/java/org/sagebionetworks/research/mpower/signup/ExternalIDRegistrationActivity.kt

```diff
@@ -0,0 +1,114 @@
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+class ExternalIDRegistrationActivity(
+        alertPresenter: ResearchAlertPresenter = ResearchAlertPresenter()) :
+        AppCompatActivity(), TextWatcher, AlertPresenter by alertPresenter {
+
+    // We create the viewModel using lazy delegate.
+    // That’s the defined approach by Google to get the ViewModel.
+    private val externalIDViewModel: ExternalIDRegistrationViewModel by lazy {
+        ViewModelProviders.of(this).get(ExternalIDRegistrationViewModel::class.java)
+    }
+
+    private val viewModelObserver = Observer<Response<User>> {
+        when(it?.status) {
+            ResponseStatus.LOADING -> showLoading()
+            ResponseStatus.SUCCESS -> success(it.data)
+            ResponseStatus.ERROR -> error(it.error)
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_external_id_registration)
+
+        externalIDViewModel.result.observe(this, viewModelObserver)
+
+        externalIdEditText.addTextChangedListener(this)
+        nameEditText.addTextChangedListener(this)
+        nextButton.setOnClickListener { nextButtonClicked() }
+    }
+
+    override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+        // no-op see afterTextChanged fun
+    }
+
+    override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+        // no-op see afterTextChanged fun
+    }
+
+    override fun afterTextChanged(s: Editable?) {
+        nextButton.isEnabled =
+                !TextUtils.isEmpty(externalIdEditText.text) &&
+                !TextUtils.isEmpty(nameEditText.text)
+    }
+
+    fun showLoading() {
+        progressBar.visibility = View.VISIBLE
```

Comment:
As for ProgressDialog being deprecated, the standard progress bar seems to be a good equivalent.  We can also explore pull to refresh at the top, but that doesn't really apply to all the screens.